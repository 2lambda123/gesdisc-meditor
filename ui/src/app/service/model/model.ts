/**
 * Model Editor API
 * No description provided (generated by Swagger Codegen https://github.com/swagger-api/swagger-codegen)
 *
 * OpenAPI spec version: 0.0.1
 * 
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 */
import { ModelIcon } from './modelIcon';


export interface Model {
    /**
     * Name of the model
     */
    name: string;
    /**
     * Model's description
     */
    description: string;
    icon: ModelIcon;
    /**
     * Model's schema
     */
    schema: string;
    /**
     * Model layout based on Angular JSON Schema form
     */
    layout: string;
    /**
     * Property name in the Model's schema for using as title of the document instance of the model
     */
    titleField?: string;
    /**
     * Any documentation (text) for the Model
     */
    documentation?: string;
    tag?: Array<string>;
}
