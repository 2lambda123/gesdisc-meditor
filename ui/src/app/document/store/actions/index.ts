export * from './history.actions';
export * from './document.actions';