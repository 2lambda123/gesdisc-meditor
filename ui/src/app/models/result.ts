export class Result {
	constructor(
		public name: string,
		public author: string,
		public modifiedOn: string
	) {}
}
