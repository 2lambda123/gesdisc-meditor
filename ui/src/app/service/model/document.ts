/**
 * Model Editor API
 * No description provided (generated by Swagger Codegen https://github.com/swagger-api/swagger-codegen)
 *
 * OpenAPI spec version: 0.0.1
 * 
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 */
import { SearchresultXmeditor } from './searchresultXmeditor';


export interface Document {
    xMeditor?: SearchresultXmeditor;
    schema?: string;
    layout?: string;
    doc?: string;
}
