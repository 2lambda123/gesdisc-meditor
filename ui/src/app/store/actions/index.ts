export * from './model.actions';
export * from './notification.actions';