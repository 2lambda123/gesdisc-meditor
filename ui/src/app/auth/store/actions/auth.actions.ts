import { Action } from '@ngrx/store';

export enum AuthActionTypes {
  GetUser = '[Auth] GetUser',
  Logout = '[Auth] Logout',
  OpenLoginDialog = '[Auth] Open Login Dialog',
  CloseLoginDialog = '[Auth] Close Login Dialog',
  LoginSuccess = '[Auth] Login Success',
  LoginFailure = '[Auth] Login Failure',
  LoginRedirect = '[Auth] Login Redirect',
}

export class GetUser implements Action {
  readonly type = AuthActionTypes.GetUser;
}

export class LoginSuccess implements Action {
  readonly type = AuthActionTypes.LoginSuccess;

  constructor(public payload: Object) {}
}

export class LoginFailure implements Action {
  readonly type = AuthActionTypes.LoginFailure;

  constructor(public payload: any) {}
}

export class LoginRedirect implements Action {
  readonly type = AuthActionTypes.LoginRedirect;
}

export class OpenLoginDialog implements Action {
  readonly type = AuthActionTypes.OpenLoginDialog;
}

export class CloseLoginDialog implements Action {
  readonly type = AuthActionTypes.CloseLoginDialog;
}

export class Logout implements Action {
  readonly type = AuthActionTypes.Logout;
}

export type AuthActions =
  | GetUser
  | LoginSuccess
  | LoginFailure
  | LoginRedirect
  | OpenLoginDialog
  | CloseLoginDialog
  | Logout;
