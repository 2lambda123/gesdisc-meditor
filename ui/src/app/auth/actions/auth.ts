import { Action } from '@ngrx/store';
import { User, Authenticate } from '../models/user';

export enum AuthActionTypes {
  GetUser = '[Auth] GetUser',
  Logout = '[Auth] Logout',
  LoginSuccess = '[Auth] Login Success',
  LoginCallback = '[Auth] Login Callback',
  LoginFailure = '[Auth] Login Failure',
  LoginRedirect = '[Auth] Login Redirect',
}

export class GetUser implements Action {
  readonly type = AuthActionTypes.GetUser;
}

export class LoginSuccess implements Action {
  readonly type = AuthActionTypes.LoginSuccess;

<<<<<<< HEAD
  constructor(public payload: Object) {}
=======
  constructor(public payload: User) {}
}

export class LoginCallback implements Action {
  readonly type = AuthActionTypes.LoginCallback;

  constructor(public payload: string) {}
>>>>>>> 58c3fd25434c01efed8927bc8045b7a611cd1612
}

export class LoginFailure implements Action {
  readonly type = AuthActionTypes.LoginFailure;

  constructor(public payload: any) {}
}

export class LoginRedirect implements Action {
  readonly type = AuthActionTypes.LoginRedirect;
}

export class Logout implements Action {
  readonly type = AuthActionTypes.Logout;
}

export type AuthActions =
  | GetUser
  | LoginSuccess
  | LoginCallback
  | LoginFailure
  | LoginRedirect
  | Logout;
