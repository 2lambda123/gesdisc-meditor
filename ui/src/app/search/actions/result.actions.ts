import { Action } from '@ngrx/store';
import { DocCatalogEntry } from '../../service/model/docCatalogEntry';

export enum ResultActionTypes {
	Search = '[Result] Search',
	SearchComplete = '[Result] Search Complete',
	SearchError = '[Result] Search Error',
	ClearResults ='[Result] Clear Results'
}

/**
 * Every action is comprised of at least a type and an optional
 * payload. Expressing actions as classes enables powerful
 * type checking in reducer functions.
 *
 * See Discriminated Unions: https://www.typescriptlang.org/docs/handbook/advanced-types.html#discriminated-unions
 */
export class Search implements Action {
	readonly type = ResultActionTypes.Search;

	constructor(public payload: string) {}
}

export class SearchComplete implements Action {
	readonly type = ResultActionTypes.SearchComplete;

	constructor(public payload: DocCatalogEntry[]) {}
}

export class SearchError implements Action {
	readonly type = ResultActionTypes.SearchError;

	constructor(public payload: string) {}
}

export class ClearResults implements Action {
	readonly type = ResultActionTypes.ClearResults;
}

/**
 * Export a type alias of all actions in this action group
 * so that reducers can easily compose action types
 */
export type ResultActionsUnion =
	| Search
	| SearchComplete
	| SearchError
	| ClearResults;
