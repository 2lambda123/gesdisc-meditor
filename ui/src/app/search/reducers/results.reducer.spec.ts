import { reducer } from './results.reducer';
import * as fromSearch from './results.reducer';
import { Search, SearchComplete, SearchError, ClearResults } from '../actions/result.actions';
import { Document } from '../../service/model/document';

describe('DocumentsReducer', () => {

	const document1 = { title: 'testDocument' };
	const document2 = { ...document1, name: '222' };
	const document3 = { ...document2, name: '333' };
	const initialState: fromSearch.State = {
		ids: [document1.title, document2.name],
		entities: {
			[document1.title]: document1,
			[document2.title]: document2,
		},
		selectedDocumentId: '',
	};

	describe('undefined action', () => {
		it('should return the default state', () => {
			const result = reducer(undefined, {} as any);

			expect(result).toMatchSnapshot();
		});
	});

	describe('SearchComplete', () => {
		function noExistingModels(
			action: any,
			documentsInitialState: any,
			initialState: any,
			documents: Document[]
		) {
			const createAction = new action(documents);

			const result = reducer(documentsInitialState, createAction);

			expect(result).toMatchSnapshot();
		}

		function existingModels(action: any, initialState: any, documents: Document[]) {
			// should not replace existing documents
			const differentDocument2 = { ...documents[0], foo: 'bar' };
			const createAction = new action([documents[1], differentDocument2]);

			const expectedResult = {
				ids: [...initialState.ids, documents[1].title],
				entities: {
					...initialState.entities,
					[documents[1].title]: documents[1],
				},
				selectedModelId: null,
			};

			const result = reducer(initialState, createAction);

			expect(result).toMatchSnapshot();
		}

		it('should add all documents in the payload when none exist', () => {
			noExistingModels(SearchComplete, fromSearch.initialState, initialState, [
				document1,
				document2,
			]);

		});

		it('should add only new documents when documents already exist', () => {
			existingModels(SearchComplete, initialState, [document2, document3]);
		});
	});

	// describe('Select Model', () => {
	// 	it('should set the selected model id on the state', () => {
	// 		const action = new SelectModel(model1.name);

	// 		const result = reducer(initialState, action);

	// 		expect(result).toMatchSnapshot();
	// 	});
	// });

	// describe('Selectors', () => {
	// 	describe('getSelectedId', () => {
	// 		it('should return the selected id', () => {
	// 			const result = fromModels.getSelectedId({
	// 				...initialState,
	// 				selectedModelId: model1.name,
	// 			});

	// 			expect(result).toMatchSnapshot();
	// 		});
	// 	});
	// });
});
