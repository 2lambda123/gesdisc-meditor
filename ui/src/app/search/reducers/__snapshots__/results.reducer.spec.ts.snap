// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DocumentsReducer SearchComplete should add all documents in the payload when none exist 1`] = `
Object {
  "entities": Object {
    "testDocument": Object {
      "title": "testDocument",
    },
  },
  "ids": Array [
    "testDocument",
  ],
  "selectedDocumentId": "",
}
`;

exports[`DocumentsReducer SearchComplete should add only new documents when documents already exist 1`] = `
Object {
  "entities": Object {
    "testDocument": Object {
      "name": "222",
      "title": "testDocument",
    },
  },
  "ids": Array [
    "testDocument",
    "222",
  ],
  "selectedDocumentId": "",
}
`;

exports[`DocumentsReducer undefined action should return the default state 1`] = `
Object {
  "entities": Object {},
  "ids": Array [],
  "selectedDocumentId": "",
}
`;
